gateway-developer-plugin/src/main/java/com/ca/apim/gateway/cagatewayconfig/BuildEnvironmentBundleTask.java

```diff
@@ -66,23 +71,38 @@ DirectoryProperty getConfigFolder() {
     public void perform() {
         final EnvironmentBundleCreator environmentBundleCreator = getInstance(EnvironmentBundleCreator.class);
         final List<File> metaDataFiles = collectFiles(into.getAsFile().get().getPath(), YML_EXTENSION);
-        if(metaDataFiles.isEmpty()) {
+        if (metaDataFiles.isEmpty()) {
             throw new MissingEnvironmentException("Metadata file does not exist.");
         }
+
         metaDataFiles.stream().forEach(metaDataFile-> {
             final Pair<String, Map<String, String>> bundleEnvironmentValues = environmentConfigurationUtils.parseBundleMetadata(metaDataFile, configFolder.getAsFile().get());
             if (null != bundleEnvironmentValues) {
-                final String bundleFileName = bundleEnvironmentValues.getLeft() + "." + configName.get() + ".environment.bundle";
+                final String envBundleFilename = getEnvBundleFileName(bundleEnvironmentValues.getLeft());
                 environmentBundleCreator.createEnvironmentBundle(
                         bundleEnvironmentValues.getRight(),
                         into.getAsFile().get().getPath(),
                         into.getAsFile().get().getPath(),
                         EMPTY,
                         PLUGIN,
-                        bundleFileName,
+                        envBundleFilename, // Passing envBundleFilename
                         bundleEnvironmentValues.getLeft()
                 );
             }
         });
     }
+
+    private String getEnvBundleFileName(String deployBundleName) {
+        final String extension = "env" + DocumentFileUtils.INSTALL_BUNDLE_EXTENSION;
+        if (configName != null && StringUtils.isNotBlank(configName.get())) {
+            return deployBundleName + "-" + removeAllSpecialChars(configName.get()) + extension;
+        } else {
+            String configFolderName = configFolder != null ? configFolder.getAsFile().get().getName() : "";
+            if (StringUtils.equalsIgnoreCase(configFolderName, "config")) {
+                return deployBundleName + "-" + extension;
```

Comment:
if configName is config, why are we ignoring it, if user is giving that as part build file, we should not ignore that.
this else block is not required, if user is not providing configName in build file, default value is EMPTY

Reply:
@RaghuReddy Here is the logic being implemented:
1) If `configName` is present, use it after removing all the special chars and append `env.{install|delete}.bundle`.
2) If `configName` is not present, use the `configFolder` name with below conditions
      2.1) If `configFolder` name is  default folder `config`, just use `-env.{install|delete}.bundle`
      2.2) Else use `configFolder` name after removing all the special chars and append `env.{install|delete}.bundle`

Reply:
@nikhilahuja , just think that if configName is not present in build file, EMPTY string comes.
Do you still need these checks?

Reply:
@RaghuReddy Yes. in that case, we need to derive config name from folder name. If folder is also the default one, then we have to just use `-env.<extension>`